src/ir/comp.rs

```diff
@@ -22,8 +22,8 @@ impl Context {
     }
 
     /// Add a new component to the context
-    pub fn comp(&mut self, is_ext: bool) -> CompIdx {
-        let comp = Component::new(is_ext);
+    pub fn comp(&mut self, is_ext: bool, is_entry: bool) -> CompIdx {
```

Comment:
Why is this needed? The existence of `is_entry` indicates that the component is an externally visible one.